src/PrimeSummation.java

```diff
@@ -0,0 +1,122 @@
+    private static int totalRecursions = 0;
+    //private static List<List<Integer>> results = new ArrayList<>();
+    //private static ArrayList<Integer> resultsSize =  new ArrayList<>();
+    
+
+    public static void initializePrimes(String fileName, int limit) {
+        
+        //private function to initialize the list from a given .csv file. Will need to come back to it based on what we change for our primeList
+        primes = new ArrayList<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+            String line;
+            while((line=br.readLine()) != null) {
+                String[] tokens = line.trim().split("\\s");
+                for (String token : tokens) {
+                    int prime = Integer.parseInt(token);
+                    if(prime <= limit) {
+                        primes.add(prime);
+                    }
+                }  
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    //The computation for the prime summation of a given number
+    /*  public static List<List<Integer>> primeSum(int num, int depth) {
+        List<List<Integer>> results = new ArrayList<>();
+        for (int i = 0; i< primes.size(); i++) {
+            int prime1 = primes.get(i);
+            if (prime1 >= num) break;
+            for (int j = i; j< primes.size(); j++) {
+                int prime2 = primes.get(j);
+                if (prime1 + prime2 > num) break;
+                if (prime1 + prime2 == num) {
+                    List<Integer> result = new ArrayList<>();
+                    result.add(prime1);
+                    result.add(prime2);
+                    results.add(result);
+                }
+            }
+        }
+        return results;
+    }*/
+    
+    public static PrimeSumResult primeSum(int num, int depth) {
+        List<List<Integer>> results = new ArrayList<>();
+        List<Integer> resultsSize = new ArrayList<>();
+        
+        for (int prime : primes) {
+            if (prime >= num) {
+                break;
+            }
+            int complement = num - prime;
+            if (primes.contains(complement)) {
+                List<Integer> result = new ArrayList<>();
+                result.add(prime);
+                result.add(complement);
+                results.add(result);
+                resultsSize.add(results.size());
+                // Check if the complement is not 2 or 3 to continue recursion
+                if (complement > 3) {
+                    totalRecursions++;
+                    PrimeSumResult recursiveResult = primeSum(complement, depth + 1);
+                    results.addAll(recursiveResult.results);
+                    resultsSize.addAll(recursiveResult.resultsSize);
+                }
+            }
+        }
+        return new PrimeSumResult(results, resultsSize);
+    }
+        
+   
+
+
+
+    public static void main(String[] args) {
```

Comment:
I dont think a main method will be used for it, instead it will be instantiated and called through the api. This is good for temporary testing though.

Reply:
Yeah that was included just to test the computation, left it there for ease in testing functionality on your end. Once we have it instantiated, will get rid of it.